### DISCUSSION TITLE
Difference between cascade='all' and cascade='all, delete'?

### DISCUSSION THREAD

**Discussion by tomasr8** [sqlalchemy/sqlalchemy#8778] [Usage Questions] [answered]
The [cascade docs](https://docs.sqlalchemy.org/en/14/orm/cascades.html) state that `all` is a synonym for `save-update, merge, refresh-expire, expunge, delete`. Why then do the docs use `cascade='all, delete'` in relationship definitions on multiple occasions? Shouldn't `cascade='all'` be sufficient? Did I miss something in the docs? Thanks!

**Comment by zzzeek** [comment, marked as answer]
those documentation sections are discussing the "delete" cascade directly, however if you were to say "cascade='delete'", that would cancel all the other cascades.  so instead of talking about "the delete cascade" and then showing examples that say "cascade='all'", the "delete" keyword is added for explicitness.

in practice, "delete" cascade is usually combined with "delete-orphan" so the typical alternate setting, as mentioned in that page, is "all, delete-orphan".

**Reply by tomasr8** [reply]
Thanks a lot! That clears it up :)
